Price carbon

FlorianLeblancDr opened the pull request:
Same as defined and discussed in #38

danielhuppmann commented:
This is basically a copy of #38 (which I started 18 months ago) so pinging @orichters & @robertpietzcker again.

danielhuppmann commented on definitions/variable/macro-economy/price.yaml:
```diff
@@ -0,0 +1,26 @@
+    description: Price of carbon (as reported by the model)
+    unit: USD_2010/t CO2
+    region-aggregation:
+      - Price|Carbon (mean):
```
```suggestion
      - Price|Carbon [Mean]:
```

danielhuppmann commented on definitions/variable/macro-economy/price.yaml:
```diff
@@ -0,0 +1,26 @@
+    region-aggregation:
+      - Price|Carbon (mean):
+          method: mean
+      - Price|Carbon (weighted by Emissions|CO2):
```
```suggestion
      - Price|Carbon [weighted by Emissions|CO2]:
```

danielhuppmann commented on definitions/variable/macro-economy/price.yaml:
```diff
@@ -0,0 +1,26 @@
+      - Price|Carbon (weighted by Emissions|CO2):
+          weight: Emissions|CO2
+          drop_negative_weights: false
+      - Price|Carbon (weighted by Final Energy):
```
```suggestion
      - Price|Carbon [weighted by Final Energy]:
```

danielhuppmann commented on definitions/variable/macro-economy/price.yaml:
```diff
@@ -0,0 +1,26 @@
+      - Price|Carbon (weighted by Final Energy):
+          weight: Final Energy
+    check-aggregate: false
+- Price|Carbon (mean):
```
```suggestion
- Price|Carbon [Mean]:
```

danielhuppmann commented on definitions/variable/macro-economy/price.yaml:
```diff
@@ -0,0 +1,26 @@
+    description: Price of carbon (computed as unweighted mean across regions)
+    unit: USD_2010/t CO2
+    skip-region-aggregation: true
+- Price|Carbon (weighted by Emissions|CO2):
```
```suggestion
- Price|Carbon [weighted by Emissions|CO2]:
```

danielhuppmann commented on definitions/variable/macro-economy/price.yaml:
```diff
@@ -0,0 +1,26 @@
+    skip-region-aggregation: true
+    note: Regional CO2 emissions can turn negative at different points in time, which
+          can result in counter-intuitive timeseries when used as weights for averages
+- Price|Carbon (weighted by Final Energy):
```
```suggestion
- Price|Carbon [weighted by Final Energy]:
```

danielhuppmann commented on definitions/variable/macro-economy/price.yaml:
```diff
@@ -0,0 +1,26 @@
+    description: Price of carbon (weighted by regional CO2 emissions)
+    unit: USD_2010/t CO2
+    skip-region-aggregation: true
+    note: Regional CO2 emissions can turn negative at different points in time, which
+          can result in counter-intuitive timeseries when used as weights for averages
```
Use `notes`, harmonize indent and full stop at end of sentence for consistency.
```suggestion
    notes: Regional CO2 emissions can turn negative at different points in time, which
       can result in counter-intuitive timeseries when used as weights for averages.
```

robertpietzcker replied:
sounds good!
I would propose to extend the notes: 
"Regional CO2 emissions can turn negative at different points in time, which can result in counter-intuitive aggregation results when used as weights for the region-aggregation. Also, regions will high carbon prices will have lower emissions, meaning the emission-weighted average of carbon prices will be low, although the high carbon price may be applied to a large share of the global energy system. Using final energy as weight overcomes these aggregation issues."

which points to the alternative without these problems, but still doesn't say "take this as default".

danielhuppmann reviewed (commented):
A few suggestions for following a consistent style.

robertpietzcker commented on definitions/variable/macro-economy/price.yaml:
```diff
@@ -0,0 +1,26 @@
+- Price|Carbon:
```
I am wondering - is it useful to have a "pure" variable without explicit region aggregation? 
I fear this could lead to constant need for questions back to the modelers "how did you aggregate this", and I am not sure we gain anything? 

also, what would the scenario explorer do for the automatic aggregation - doesn't it need a specification which of the following variants is applied when reading in "Price|Carbon"? 

Maybe I overlook something, but otherwise I would propose to drop this general term and only have the three explicit versions below.

robertpietzcker replied:
Ah, so teams upload with "Price|Carbon", and then in the explorer one will see "Price|Carbon [Mean]" (or whatever was decided as aggregation mechanism by the project lead?

FlorianLeblancDr commented:
@robertpietzcker I just added your suggested notes with some rephrasing. 
Is it ok like this?
Should we split the note between the two variable (weighted by emissions and by final energy?). This would avoid long notes.

robertpietzcker commented on definitions/variable/macro-economy/price.yaml:
```diff
@@ -0,0 +1,31 @@
+       can result in counter-intuitive aggregation results when used as weights for
+       the region-aggregation. Also the [weighted by Final Energy] methods is
+       complementary and overcomes other aggregation issues - regions with high carbon
+       prices can be associated with low emission levels, meaning the its contribution
```
meaning its contribution (delete the "the")

robertpietzcker reviewed (approved):
thanks - looks good to me!

robertpietzcker commented:
> Should we split the note between the two variable (weighted by emissions and by final energy?). This would avoid long notes.

I have to admit I don't know if there is a preferred approach - both would be fine for me.

FlorianLeblancDr commented:
> > Should we split the note between the two variable (weighted by emissions and by final energy?). This would avoid long notes.
> 
> I have to admit I don't know if there is a preferred approach - both would be fine for me.

Let's keep it like this then.
@danielhuppmann This seems ready to be merged.

FlorianLeblancDr commented:
ping @danielhuppmann  Should we merge this?

danielhuppmann reviewed (approved):
Thank you, looks good
